reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -2010,7 +2010,9 @@ public T block(Duration timeout) {
 	 * @param afterSuccessOrError the callback to call after {@link Subscriber#onNext}, {@link Subscriber#onComplete} without preceding {@link Subscriber#onNext} or {@link Subscriber#onError}
 	 *
 	 * @return a new {@link Mono}
+	 * @deprecated prefer using {@link #doAfterTerminate(Runnable)}. will be removed in 3.4.0
```

Comment:
```suggestion
	 * @deprecated prefer using {@link #doAfterTerminate(Runnable)} and/or {@link #doOnError(Consumer)}. will be removed in 3.4.0
```

Reply:
actually, do OnError is not a suitable alternative as it acts BEFORE propagating the onError. but doFinally is fine

Reply:
I adapted the suggestion accordingly :+1:

Reply:
but `doAfterTerminate`/`doFinally` do not provide access to the error (see https://github.com/reactor/reactor-core/pull/1868#discussion_r318665450 )

Reply:
but `doFinally` does not expose the error, only the signal type, and the previous operator was providing it (see this comment: https://github.com/reactor/reactor-core/pull/1868#discussion_r318665450 )

Reply:
Ok, after discussing in Slack, it sounds fine to me since `doAfterError` makes little to none sense 👍